### PR TITLE
DEX-1092 Update version of sttp-client

### PR DESCRIPTION
**Pull request by ngadiyak**

### CODE REVIEW

File: dex-it-common/src/main/scala/com/wavesplatform/dex/it/sttp/ResponseParsers.scala [hunk 1]
```diff
@@ -1,29 +1,33 @@
...
 import com.google.common.primitives.Longs
-import com.softwaremill.sttp.{DeserializationError, ResponseAs, MonadError => _, _}
 import com.typesafe.config.{Config, ConfigFactory}
 import play.api.libs.json.JsError
+import sttp.client3._
+import sttp.client3.playJson._
 
 import scala.util.{Failure, Success, Try}
 
 object ResponseParsers {
```
Comment (vsuharnikov) [inline dex-it-common/src/main/scala/com/wavesplatform/dex/it/sttp/ResponseParsers.scala:11]: This class is not used, looks like we can remove it.

**Review by vsuharnikov — CHANGES_REQUESTED** [review]

File: dex-it-common/src/main/scala/com/wavesplatform/dex/it/api/AsyncEnrichedApi.scala [hunk 2]
```diff
@@ -3,33 +3,32 @@ package com.wavesplatform.dex.it.api
...
 
   type R[EntityT] = Future[EnrichedResponse[ErrorT, EntityT]]
 
-  def mk[EntityT: Reads](req: Request[String, Nothing]): R[EntityT] =
-    httpBackend
-      .send[String](req.tag("requestId", UUID.randomUUID))
-      .map(EnrichedResponse(_, new EnrichedResponse.AsJson[ErrorT, EntityT]))
+  def mk[EntityT: Reads](req: Request[Either[String, String], Any]): R[EntityT] = req
+    .tag("requestId", UUID.randomUUID)
+    .contentType("application/json")
```
Comment (vsuharnikov) [inline dex-it-common/src/main/scala/com/wavesplatform/dex/it/api/AsyncEnrichedApi.scala:21]: This is not required, probably wrong in some cases. Content-Type is specified by BodySerializer

File: dex/src/main/scala/com/wavesplatform/dex/tool/connectors/RestConnector.scala [hunk 3]
```diff
@@ -5,28 +5,29 @@ import com.wavesplatform.dex.cli.ErrorOr
...
 }
 
 object RestConnector {
 
   type ErrorOrJsonResponse = ErrorOr[JsValue]
-  type RequestFunction = RequestT[Empty, Either[String, String], Nothing] => Request[Either[String, String], Nothing]
+  type RequestFunction = RequestT[Empty, Either[String, String], Any] => Request[Either[String, String], Any]
 
-}
+}
```
Comment (vsuharnikov) [inline dex/src/main/scala/com/wavesplatform/dex/tool/connectors/RestConnector.scala:33]: newline

**Review by vsuharnikov — CHANGES_REQUESTED** [review]

File: dex-it-common/src/main/scala/com/wavesplatform/dex/it/api/AsyncEnrichedApi.scala (hunk 2 above)
Comment (ngadiyak) [reply dex-it-common/src/main/scala/com/wavesplatform/dex/it/api/AsyncEnrichedApi.scala:21]: fixed

File: dex/src/main/scala/com/wavesplatform/dex/tool/connectors/RestConnector.scala (hunk 3 above)
Comment (ngadiyak) [reply dex/src/main/scala/com/wavesplatform/dex/tool/connectors/RestConnector.scala:33]: fixed

**Comment by vsuharnikov** [comment]
1. We need to wait for https://github.com/wavesplatform/matcher/pull/482
2. Then rebase your changes
3. Check all fields of MatcherHttpStatus are Working in https://github.com/wavesplatform/matcher/blob/master/dex-it-common/src/main/scala/com/wavesplatform/dex/it/docker/DexContainer.scala#L62

**Review by vsuharnikov — APPROVED** [review]

**Pull request merged**